File: python/31b_1433A_BoringApartments.py
from math import log10 as log;
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        x = int(input())
        dig, num = 0, int(log(x)) + 1
        dig = dig + (10 * ((x % 10) - 1)) + ((num * (1 + num)) // 2)
        ans.append(dig)
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/31b_1433A_BoringApartments.cpp
#include<bits/stdc++.h>

int main(){
    int t, x, dig, num;
    std::vector<int> ans;
    std::cin>>t;
    for(int i = 0; i < t; i++){
        std::cin>>x;
        dig = 0;
        dig += (10 * ((x % 10) - 1));
        num = log10(x) + 1;
        dig += ((num) * (1 + num)) / 2;
        ans.push_back(dig);
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}